### PR TITLE
Fix/timer

### PR DESCRIPTION
**Pull request by FrogTravel**
Timer bug fix

### CODE REVIEW

File: feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/data/repository/StakingRepositoryImpl.kt [hunk 1]
```diff
@@ -97,25 +97,25 @@ class StakingRepositoryImpl(
...
         binding = ::bindCurrentSlot
     )
 
-    override suspend fun genesisSlot() = localStorage.queryNonNull(
+    override suspend fun genesisSlot() = remoteStorage.queryNonNull(
         keyBuilder = { it.metadata.babe().storage("GenesisSlot").storageKey() },
         binding = ::bindCurrentSlot
     )
 
     override suspend fun eraStartSessionIndex(currentEra: BigInteger): BigInteger {
```
Comment (ef1rspb) [inline feature-staking-impl/src/main/java/jp/co/soramitsu/feature_staking_impl/data/repository/StakingRepositoryImpl.kt:115]: `activeEra: EraIndex`

File: feature-staking-api/src/main/java/jp/co/soramitsu/feature_staking_api/domain/api/StakingRepository.kt [hunk 2]
```diff
@@ -22,7 +23,7 @@ interface StakingRepository {
 
     suspend fun genesisSlot(): BigInteger
 
-    suspend fun eraStartSessionIndex(currentEra: BigInteger): BigInteger
+    suspend fun eraStartSessionIndex(currentEra: BigInteger): EraIndex
```
Comment (ef1rspb) [inline feature-staking-api/src/main/java/jp/co/soramitsu/feature_staking_api/domain/api/StakingRepository.kt:26]: activeEra: EraIndex

File: feature-staking-api/src/main/java/jp/co/soramitsu/feature_staking_api/domain/api/StakingRepository.kt [hunk 3]
```diff
@@ -73,12 +76,12 @@ interface StakingRepository {
...
 suspend fun StakingRepository.getActiveElectedValidatorsExposures() = electedExposuresInActiveEra.first()
 
-suspend fun StakingRepository.historicalEras(countRange: (start: Int, finish: Int) -> IntRange): List<BigInteger> {
+suspend fun StakingRepository.historicalEras(): List<BigInteger> {
     val activeEra = getActiveEraIndex().toInt()
     val currentEra = getCurrentEraIndex().toInt()
     val historyDepth = getHistoryDepth().toInt()
 
-    val historicalRange = countRange((currentEra - historyDepth), activeEra)
+    val historicalRange = (currentEra - historyDepth) until activeEra
```
Comment (ef1rspb) [inline feature-staking-api/src/main/java/jp/co/soramitsu/feature_staking_api/domain/api/StakingRepository.kt:84]: swift implementation:
```swift
struct ChainHistoryRange {
    let currentEra: EraIndex
    let activeEra: EraIndex
    let historyDepth: UInt32

    var erasRange: [EraIndex] {
        Array(currentEra - historyDepth ... activeEra - 1)
    }
}
```

Comment (FrogTravel) [reply feature-staking-api/src/main/java/jp/co/soramitsu/feature_staking_api/domain/api/StakingRepository.kt:84]: It is the same =)

Comment (ef1rspb) [reply feature-staking-api/src/main/java/jp/co/soramitsu/feature_staking_api/domain/api/StakingRepository.kt:84]: Sure)

**Pull request closed without merging**